frontend/views/pages/Contributions.vue

```diff
@@ -241,17 +241,22 @@ export default {
     openModal (modal) {
       sbp('okTurtles.events/emit', OPEN_MODAL, modal)
     },
-    async handleNonMonetary (type, value) {
+    async handleNonMonetary (type, value, done) {
+      console.log('handle new-value')
       try {
         const groupProfileUpdate = await sbp('gi.contracts/group/groupProfileUpdate/create',
           { [type]: value },
           this.$store.state.currentGroupId
         )
         await sbp('backend/publishLogEntry', groupProfileUpdate)
+        // TODO/BUG - Discover why "value" gets added to the list
+        // but Contribution input takes longer to unmount. (in Slow 3G).
```

Comment:
@taoeffect see the [video recordings I left on Slack](https://okturtles.slack.com/files/U9SGRFKT6/FUJS6KSE5/inspect_flow.mp4) (on a thread).

This is a Slow 3G flow of adding one contribution and adding a second contribution with logs strategically placed before and after the `this.$emit`, on `handleNonMonetary` and `beforeUpdate`.
Compare the logs with their placement in the code (I pushed the logs).

So, it seems the state is updated in the moment `events/emit GI Message` is processed. In this moment, the components received the new list of contributions. But we still need to process `settings/save user-1. Meanwhile, the `Contribution.vue` shows that "buggy state" because `done()` wasn't called yet.  Oonly after `settings/save` is completed, that `done()` gets called.

What if we saved the user settings before processing the GI Message? That way the moment the state updates, there are no more network requests to make and we don't risk showing incorrect states.

Reply:
@sandrina-p I'm not seeing this as a problem that has anything to do with `'gi.db/settings/save'` or `done()`.

`'gi.db/settings/save'` does not affect the UI at all. All it does is save the latest state to localStorage. It doesn't impact the UI in any way besides the loading of the state the next time the user logs in or refreshes the page.

The `done()` callback seems to have been removed in a recent commit (except in `AddMembers`, where I think it also should be removed), so that's no longer relevant here either.

AFAICT, the issue seems to be that in `Contribution.vue`, `contributionsList` (which is bound to `ourGroupProfile.nonMonetaryContributions`), is updated, and the input form is still shown.

So because `contributionsList` is updated, and the form is still shown, `vuelidate` seems to get triggered, and it evaluates the form, sees that the same input is being added to the list, and shows the error.

How to solve this? I'm not entirely sure. I can see at least two different approaches:

1. Delay vuelidate's evaluation by manually using `$touch()` and `$dirty()` at the appropriate times (see [Without v-model](https://vuelidate.js.org/#sub-without-v-model) section of vuelidate docs)
2. Immediately hide the form after submit is hit (as long as the form is valid), so that the vuelidate error isn't visible (this might be the simpler approach, so try it first)

Reply:
> 'gi.db/settings/save' does not affect the UI at all. 

I didn't say it changed the UI. I'm saying it cause the click handler to take longer to resolve, so `Contribution` can hide...

1. No, that's a workaround solution that doesn't solve the root of the problem: Why isn't the click handler resolved as soon as the state gets updated.
2. This is our current approach (in master), without using loading state. This kind of approach has a name: ["Optimistic UI"](https://www.smashingmagazine.com/2016/11/true-lies-of-optimistic-user-interfaces/).

But this still doesn't explain why we have a delay between when the state is updated and when the click handler resolves. If you see the Network, there's a request being made during that period. That request is what's preventing the handler to finish as soon as the state is updated. And that request doesn't seem necessary because the state was already updated. So what else are we waiting for that is delaying the click resolve.

Reply:
Here's a GIF showing the extra request that takes time and it's preventing the click handler to resolve.
![flow_net2](https://user-images.githubusercontent.com/14869087/76102715-4f7e5a00-5fc8-11ea-8267-0c914077981d.gif)

Reply:
Shower thought: realized there's a simple fix: clear the input field as soon as the return key is pressed (and validations pass).

Reply:
What if there's an error? Then the user would need to add the text again... Another downside is while it's submitting the user doesn't see the text what can be strange.

Reply:
> What if there's an error

There is no error because it's cleared after the validations have passed. This seems like the right solution.

Reply:
An error related to the server might happen... And this isn't the right solution because this "state / request order issue" might happen in other scenarios too. I'll create the issue about it.

Reply:
> An error related to the server might happen...

That's OK. That will still appear and be visible.

Reply:
If we clean the input, an inline error will still appear saying the field is required.

I think we should move on as it is. This issue is bigger than this Contribution scenario. I'm sure it happens in other places. We just didn't detect it yet.

Issue opened: https://github.com/okTurtles/group-income-simple/issues/858

Reply:
@taoeffect, as discussed in our weekly video call, I changed the code to clean (reset) the input when submitting it. It doesn't solve the issue, but now at least the user doesn't see any "wrong " state nor inline error. Still, I think we should understand what's the purpose of that 2º network requests and if we can avoid it...